### PR TITLE
Merge current status of geometry module

### PR DESCRIPTION
**Pull request by vhirtham**

### CODE REVIEW

**Comment by vhirtham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test_geometry.py  1
- mypackage/geometry.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/volker.hirthammer/pythonTest/pullRequest?prid=4713648&bid=15743544)

**Comment by vhirtham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test_geometry.py  1
- mypackage/geometry.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/volker.hirthammer/pythonTest/pullRequest?prid=4713648&bid=15743544)

**Comment by vhirtham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test_geometry.py  1
- mypackage/geometry.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/volker.hirthammer/pythonTest/pullRequest?prid=4713648&bid=15743544)

**Comment by vhirtham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test_geometry.py  1
- mypackage/geometry.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/volker.hirthammer/pythonTest/pullRequest?prid=4713648&bid=15743544)

**Comment by vhirtham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test_geometry.py  1
- mypackage/geometry.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/volker.hirthammer/pythonTest/pullRequest?prid=4713648&bid=15743544)

File: tests/test_geometry.py [hunk 1]
```diff
@@ -2,14 +2,64 @@
...
+    # Invalid center point
+    with pytest.raises(ValueError):
+        geo.Shape2D([0, 0], [1, 1], geo.Shape2D.ArcSegment([0, 1.1]))
+
+    shape = geo.Shape2D([0, 0], [1, 1], segment=geo.Shape2D.ArcSegment([0, 1]))
+    shape.add_segment([2, 2], segment=geo.Shape2D.ArcSegment([2, 1]))
+    # Invalid center point
+    with pytest.raises(ValueError):
+        shape.add_segment([3, 1], segment=geo.Shape2D.ArcSegment([2.1, 1]))
+
```
Comment (CagtayFabry) [inline tests/test_geometry.py]: should be moved to end of test

Comment (vhirtham) [reply tests/test_geometry.py]: Done

**Comment by vhirtham** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test_geometry.py  1
- mypackage/geometry.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/volker.hirthammer/pythonTest/pullRequest?prid=4713648&bid=15743544)

**Pull request merged**